### PR TITLE
[Merged by Bors] - refactor(SetTheory/Cardinal/Ordinal): make `Cardinal.aleph'` an order iso

### PR DESCRIPTION
**Pull request by vihdzp**
We deprecate  `alephIdx.initialSeg`, `alephIdx`, `alephIdx.relIso`, `Aleph'.relIso`, and `aleph'Equiv`, which were just stating this idea in slightly different ways.

We also make `Cardinal.aleph` an order embedding.

---

- [x] depends on: #16222

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/SetTheory/Cardinal/Ordinal.lean [hunk 1]
```diff
@@ -209,41 +225,46 @@ theorem aleph'_omega : aleph' ω = ℵ₀ :=
...
-  uncountable cardinal, and so on. -/
-def aleph (o : Ordinal) : Cardinal :=
-  aleph' (ω + o)
+/-- The `aleph` function gives the infinite cardinals listed by their ordinal index. `aleph 0 = ℵ₀`,
+`aleph 1 = succ ℵ₀` is the first uncountable cardinal, and so on.
+
+For a version including finite cardinals, see `Cardinal.aleph'`. -/
+def aleph : Ordinal ↪o Cardinal :=
+  OrderEmbedding.ofMapLEIff
+    (fun o => aleph' (ω + o)) fun _ _=> aleph'_le.trans (add_le_add_iff_left ω)
```
Comment (eric-wieser) [inline Mathlib/SetTheory/Cardinal/Ordinal.lean:240]: Can you build this by composing `aleph'.toOrderEmbedding` with the order embedding that adds `ω`?

Comment (vihdzp) [reply Mathlib/SetTheory/Cardinal/Ordinal.lean:240]: What's the name of the latter?

**Review by eric-wieser — APPROVED** [review]

File: Mathlib/SetTheory/Cardinal/Ordinal.lean (hunk 1 above)
Comment (eric-wieser) [reply Mathlib/SetTheory/Cardinal/Ordinal.lean:240]: I don't know, but you can loogle for it using the lemma that characterizes its application.

If it doesn't exist, you could build it inline in this proof.

Comment (vihdzp) [reply Mathlib/SetTheory/Cardinal/Ordinal.lean:240]: Oh I found it, it's [`OrderEmbedding.addLeft`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Order/Monoid/Basic.html#OrderEmbedding.addLeft).

File: Mathlib/SetTheory/Cardinal/Ordinal.lean
Comment (eric-wieser) [inline Mathlib/SetTheory/Cardinal/Ordinal.lean:1]: Can you fix the module docstring to not refer to `Cardinal.aleph/idx`, which look like a typo of the deprecated name?

**Review by eric-wieser — COMMENTED** [review]
Since you requested their review,

bors d=@YaelDillies

Feel free to merge without review if you don't care much about this PR, Yael.

File: Mathlib/SetTheory/Cardinal/Ordinal.lean
Comment (vihdzp) [reply Mathlib/SetTheory/Cardinal/Ordinal.lean:1]: I've updated the docstring, and I've also removed the obsolete part about `bit0` and `bit1`.

File: Mathlib/SetTheory/Cardinal/Ordinal.lean [hunk 2]
```diff
@@ -73,6 +70,26 @@ theorem noMaxOrder {c} (h : ℵ₀ ≤ c) : NoMaxOrder c.ord.toType :=
 
 section aleph
 
+/-- The `aleph'` function gives the cardinals listed by their ordinal index. `aleph' n = n`,
+`aleph' ω = ω`, `aleph' (ω + 1) = succ ℵ₀`, etc.
```
Comment (YaelDillies) [inline Mathlib/SetTheory/Cardinal/Ordinal.lean:74]:
```suggestion
`aleph' ω = ℵ₀`, `aleph' (ω + 1) = succ ℵ₀`, etc.
```
no?

**Comment by YaelDillies** [comment]
The rest looks sensible. Thanks Eric for simplifying my job!

bors merge

**Pull request closed without merging**